osquery/logger/plugins/tests/tls_logger_tests.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <osquery/logger.h>
+#include <osquery/database.h>
+
+#include "osquery/core/test_util.h"
+
+#include "osquery/logger/plugins/tls.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+class TLSLoggerTests : public testing::Test {
+ public:
+  size_t getIndex(const std::shared_ptr<TLSLoggerPlugin>& plugin) {
+    return plugin->log_index_;
+  }
+
+  void runCheck(const std::shared_ptr<TLSLogForwarderRunner>& runner) {
```

Comment:
This little trampoline allows the `TLSLoggerTests` to be the only friend of `TLSLogForwarderRunner`. For some reason I don't quite like the `FRIEND_TEST` macros littering the class definitions. Providing an abstraction like this in the base test class allows quick addition of new tests.